### ISSUE TITLE
Firefox 57

### ISSUE THREAD

**Issue by bodeha** [Aris-t2/ClassicThemeRestorer#380]
this plugin we want, new fire looks bad omg, please rapair this man :)

**Comment by Aris-t2** [comment]
Unfortunately Mozilla decided to not support these kind of add-ons anymore, so there won't be a WebExtension or a Firefox 57+ version of this add-on. Read here about it: https://github.com/Aris-t2/ClassicThemeRestorer/issues/299

Contact Mozilla and tell them what you think about their move to drop support of the superior add-on system they had:
https://support.mozilla.org/

Use custom CSS code for Firefox 57+ ( https://github.com/Aris-t2/CustomCSSforFx/issues/2 ) to keep using classic ui features or switch to Waterfox browser ( https://www.waterfoxproject.org/  - a Firefox fork with support for legacy add-ons) to keep classic add-ons.

**Comment by ExE-Boss** [comment]
It might be possible if [this specification](https://exe-boss.tech/specifications/web-extensions/gecko-elevated/) [gets accepted](https://www.reddit.com/r/firefox/comments/6qtd47/eli5_how_does_xul_relate_to_html_is_xul_superset/dlq5gwa/?context=1) (I’ll open a bug soon, once I finish version 1.0-RC1).

**Comment by bodeha** [comment]
in waterfox this plugin not working to.

**Comment by bodeha** [comment]
but YUPI! looks same like old one :) thx i try to stay longer whitch this water project :)

**Comment by Aris-t2** [comment]
@ExE-Boss
Sounds great, but I'm wondering why all requests for creating interface or ui-customizable APIs  were declined in the past.

> What I meant is that the APIs will return, and the healthy Addon ecosystem will bring new addons, both in numbers and types of addons.

Reading this posted by a "Platform Engineer at Mozilla" makes me really hate Mozilla, if this is true. Why the ... was it so hard to provide the required APIs first and drop legacy add-on system afterwards. Giving devs time to port their add-ons for a year or so and everybody would be happy. I'm really thinking that company is not only ruled by absolute ..., but is also full of ... developers.

@bodeha 
Good to hear your figured it out.

**Comment by PandaCodex** [comment]
Cyberfox is still good

> Cyberfox 52.8 will be the last which is scheduled for 2018-05-01 and ends 2018-06-26 so 52.8 at this stage in time will be the last Cyberfox release

Still ages of time, with talks floating around that next ESR version may give extra year to Cyberfox

**Comment by Burstnok** [comment]
Wouldn't it be possible to offer the service CTR did for years via an standalone program? WebExtensions can't modify the browser UI anymore but an executable could modfiy the CustomCSS files and offer the option for an UI like CTR used to have before FF57.

**Comment by Aris-t2** [comment]
Probably, but such an application would require far more maintenance (to keep supporting all Firefox builds) than offering plain CSS code for userChrome.css. Not worth the effort.

**Comment by Aris-t2** [comment]
That was a good one.
Mozilla add thousands of changes on every build since Firefox 4 release.

**Comment by Burstnok** [comment]
oh...so the CustomCSS is only temporary? Hoped that would work for a while at least....

**Comment by Aris-t2** [comment]
I try to keep it updated.

**Comment by youtubedownloads** [comment]
* Which Web browser supports TAB HEIGHT Adjustment? ALl the tabs are too small to use an a 32 Inch screen? Can the develeopers at least allow a TAB Height Adjuster add on?

**Comment by Aris-t2** [comment]
@youtubedownloads 
Firefox 57+ can have higher tabs using CSS in userChrome.css.
Same applies to every Firefox fork and CTR can set tab height in Waterfox and Basilisk browser.

Vivaldi also allows to modify ui using custom CSS code.

**Comment by youtubedownloads** [comment]
Please post link on how to modify tab height in CSS. I would like to modify the tab height.

**Comment by Aris-t2** [comment]
Firefox 57+
1. `about:support > Profile Folder > Open Folder`
2. create a new folder called `chrome`
3. create a new text file and save it as `userChrome.css` (not _userChrome.css.txt_!)
4. add this to  `userChrome.css`
```css
:root {
  --tab-min-height: 46px !important; /* your height value */
}
```
5. Save file and restart Firefox.

**Comment by youtubedownloads** [comment]
Can't seem to open the about:support profile folder through the http: search function

I can access about:support but not directly to the profile folder.

Note*: I have a profile folder in Mozila in that folder there is a .default named Folder. inside that folder has chrome debugger folder / sessions folder / extentions folder / features folder etc. 

In the .default folder I added a : chrome folder: userChrome text file inside.

I copied:

 :root {
  --tab-min-height: 46px !important; /* your height value */
}

I have two options to save: Save as (All) Or save as .txt.

Which one should I save it as?

What is "your height value?" - Do I enter a number?------ ;/*100*/

**Comment by Aris-t2** [comment]
You only need to save the file as .css not .txt
---> userChrome.css (not userChrome.txt or userChrome.css.txt)

Default height value is 32px, CTRs squared tabs use 26px.

You have the code, so you can of course change it, the 46px was just an example, use own values.

**Comment by youtubedownloads** [comment]
The code did not do anything Tab Height Still the same? 

There is no other code in the Folder (Chrome)...just copy and pasted the above code.

Does it require any other code before it?

This should not be this complicated.

**Comment by Aris-t2** [comment]
Technically it is a text file, but some OSs hide default file endings resulting in a masked "userChrome.css.txt" file.

Post a screenshot of your "chrome" folder. Mine looks like this (note, I have move stuff there, the CustomCSSforFx project).

![1](https://user-images.githubusercontent.com/6013569/34740384-7f87bdd2-f57f-11e7-8a46-a3466428baeb.png)

**Comment by youtubedownloads** [comment]
Mine is the exact same, in the .deafult folder etc. Running Windows 7 64 bit SP1.

My Chrome Folder does not have config folder or css Folder or image folder or userchrome or xml folder...it just has the userChrome.css text in it the above do not exist in the chrome Folder I created.

Do I need to download chrome? I don't use chrome.

**Comment by Aris-t2** [comment]
Download the project from here https://github.com/Aris-t2/CustomCSSforFx/issues/2 and tell me, if its working or not.

**Comment by youtubedownloads** [comment]
* I put the code in the folder and was unable to access the add-ons menu in Firefox.

I should be able to access the add-ons and turn them on-off. The code would not display them?

Is there a menu to customize tab size and height somewhere? I could not find it.

**Comment by Aris-t2** [comment]
Looks to me like you Firefox / Firefox profile is deeply broken. The code affecting add-ons area is supposed to restore the version number. You can simply edit userContent.css for that or remove it. Restoring version number works fine for me in Firefox 57-59 on Windows 7-10, Linux and macOS.

Anyway I would suggest to create an entirely new browser profile.

**Comment by youtubedownloads** [comment]
Ok, Thanks.

Is there going to be an Official release from Firefox regarding TAB height + width Add-on? Adjusments?

I can't seem to find any addons that do this, most add-ons are just TAB trees etc.

**Comment by Aris-t2** [comment]
Of course not. ;-)

The reason why we write and use CSS code in userChrome.css and userContent.css was Mozillas decision to remove browser ui modification abilities by only allowing powerless web extensions, that can only modify web content, but not the browser ui.
The very limited stuff web extensions can achieve is editing theme colors the way default light/dark themes do it. Vivaldifox is a WE for this for example: https://addons.mozilla.org/de/firefox/addon/vivaldifox/

Anyway, you want something different. I suggest do download and install Portable Firefox and download and apply the previously mentioned CustomCSSforFx there (close default Firefox before running PF).
That way you will get a clean Firefox you can ran tests in without affecting your current setup.

**Comment by youtubedownloads** [comment]
* Ok thank you for the assistance, this has been informative. 

I will try a clean install of Portable Firefox and will see what new UI features are added using the above code etc.

* It is very disappointing Firefox does not allow simple TAB Height adjustments in the browser. That's why I stay away from Microsoft Edge browser because the tabs a too thin to waste time on.
